File: smart_travel_BE/src/main/java/com/example/smart_travel_BE/entity/DestinationImage.java
package com.example.smart_travel_BE.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "destination_images")
public class DestinationImage {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "destination_id", nullable = false)
    @JsonIgnore
    private Destination destination;

    @Column(name = "image_url", nullable = false, length = 500)
    private String imageUrl;

    @Column(name = "is_primary", nullable = false)
    private Boolean isPrimary = false;

    @Column(name = "display_order", nullable = false)
    private Integer displayOrder = 0;

    @CreationTimestamp
    @Column(name = "uploaded_at", nullable = false)
    private LocalDateTime uploadedAt;
}

File: smart_travel_FE/lib/domain/entities/destination_image.dart
import 'package:equatable/equatable.dart';

class DestinationImage extends Equatable{
  final int? id;
  final String? imageUrl;
  final bool? isPrimary;
  final int? displayOrder;
  final DateTime? uploadedAt;

  DestinationImage({this.id, this.imageUrl, this.isPrimary, this.displayOrder, this.uploadedAt});

  @override
  // TODO: implement props
  List<Object?> get props => [id, imageUrl, isPrimary, displayOrder, uploadedAt];


}